Source/DayView.swift

```diff
@@ -46,21 +54,49 @@ class DayView: UIView {
     let size = CGSize(width: contentWidth, height: 0)
     timelinePager.contentSize = size
 
-    dayHeaderView.anchorAndFillEdge(.Top, xPad: 0, yPad: 20, otherSize: headerHeight)
+    dayHeaderView.anchorAndFillEdge(.Top, xPad: 0, yPad: 0, otherSize: headerHeight)
     timelinePager.alignAndFill(align: .UnderCentered, relativeTo: dayHeaderView, padding: 0)
   }
 }
 
 extension DayView: PagingScrollViewDelegate {
   func viewRequiresUpdate(view: UIView, viewBefore: UIView) {
+    let timeline = (view as! TimelineContainer).timeline
+    timeline.date = currentDate.dateByAddingDays(1)
 
+    if let dataSource = dataSource {
+      timeline.eventViews = dataSource.eventViewsForDate(timeline.date)
+    }
   }
 
   func viewRequiresUpdate(view: UIView, viewAfter: UIView) {
+    let timeline = (view as! TimelineContainer).timeline
+    timeline.date = currentDate.dateByAddingDays(-1)
 
+    if let dataSource = dataSource {
+      timeline.eventViews = dataSource.eventViewsForDate(timeline.date)
+    }
+  }
+
+  func scrollviewDidScrollToView(view: UIView) {
+    let timeline = (view as! TimelineContainer).timeline
+    currentDate = timeline.date
+    dayHeaderView.selectDate(currentDate)
   }
 }
 
 extension DayView: DayHeaderViewDelegate {
-  func dateHeaderDateChanged(newDate: NSDate) {}
-}
\ No newline at end of file
+  func dateHeaderDateChanged(newDate: NSDate) {
+    //TODO: refactor
+    if newDate.isEarlierThan(currentDate) {
+     let timelineContainer = timelinePager.reusableViews.first! as! TimelineContainer
```

Comment:
I don't think it will add clarity to code not the fault tolerance.

I don't like that paginated scrollview, I will think about better scrolling system. In ObjC it was OK to cast back and forth, in Swift it requires a lot of exclamation points.